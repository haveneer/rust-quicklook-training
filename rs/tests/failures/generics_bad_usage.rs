fn foo<T>(arg: T) -> T { return arg + 1; }

fn main() {
    foo(1);
    foo(3.14);
    foo("Hello");
}

// Solution au moins pour foo(1)
// fn foo<T: std::ops::Add<i32, Output=T>>(arg: T) -> T { return arg + 1; }
