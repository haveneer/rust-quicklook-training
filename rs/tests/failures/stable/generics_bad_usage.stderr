error[E0369]: cannot add `{integer}` to `T`
 --> tests/failures/stable/generics_bad_usage.rs:1:37
  |
1 | fn foo<T>(arg: T) -> T { return arg + 1; }
  |                                 --- ^ - {integer}
  |                                 |
  |                                 T
  |
help: consider restricting type parameter `T`
  |
1 | fn foo<T: std::ops::Add<i32>>(arg: T) -> T { return arg + 1; }
  |         ++++++++++++++++++++
